xrpl/models/transactions/transaction.py

```diff
@@ -114,7 +147,7 @@ class Transaction(BaseModel):
     memos: Optional[List[Memo]] = None
     signers: Optional[List[Signer]] = None
     source_tag: Optional[int] = None
-    signing_pub_key: Optional[str] = None
+    signing_pub_key: str = ""
```

Comment:
allows us to avoid a bunch of checks for `None`. AFAICT it can never be `None` anyway

Reply:
This is probably OK.

The XRP Ledger _does_ make a distinction between the binary form of transactions where `SigningPubKey` is present but empty (multi-signed transactions only) vs. ones where it is absent (unsigned transaction instructions). For multi-signed transactions the field needs to be serialized as empty, and in the latter case it needs to be added with the appropriate key when you serialize the transaction for signing.